src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -329,6 +373,9 @@ class TaskBuilderTest extends MarathonSpec with Matchers {
       .build
     TextFormat.shortDebugString(networkInfos.head) should equal(TextFormat.shortDebugString(networkInfoProto))
     networkInfos.head should equal(networkInfoProto)
+
+    taskInfo.hasDiscovery should be (true)
+    taskInfo.getDiscovery.getName should be (taskInfo.getName)
```

Comment:
Don't know if I missed that, but we should have at least one test case, that tests all properties of a discovery info.